sfs/mono/drivingfunction.py

```diff
@@ -673,6 +673,7 @@ def _sph_hn2(n, z):
     return jn - 1j * yn
 
 
-def _max_order_circular_harmonics(N):
+def _max_order_circular_harmonics(N, max_order):
     """Compute order of 2D HOA."""
-    return N // 2 if N % 2 == 0 else (N - 1) // 2
+    return (max_order if max_order is not None
+            else N // 2 if N % 2 == 0 else (N - 1) // 2)
```

Comment:
Nice one! Behaviour is equivalent for integer `N` but differs for fractional. That was probably not intended so I changed it to suggestion.